File: src/go/tree/94.binary-tree-inorder-traversal.go
package main

func inorderTraversal(root *TreeNode) []int {
	res := []int{}
	var helper func(*TreeNode)
	helper = func(node *TreeNode) {
		if node == nil {
			return
		}
		helper(node.Left)
		res = append(res, node.Val)
		helper(node.Right)
	}
	helper(root)
	return res
}

File: src/python/tree/94.binary-tree-inorder-traversal.py
from typing import List, Optional
from lc_test_utils import TreeNode

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        def dfs(node):
            if not node:
                return
            dfs(node.left)
            res.append(node.val)
            dfs(node.right)
        dfs(root)
        return res
